Source/Parallelization/WarpXComm.cpp

```diff
@@ -41,43 +41,16 @@ WarpX::UpdateAuxilaryDataStagToNodal ()
             "WarpX::UpdateAuxilaryDataStagToNodal: PSATD solver requires "
             "WarpX build with spectral solver support.");
     }
-#else
-    amrex::Gpu::DeviceVector<Real> d_stencil_coef_x;
-    amrex::Gpu::DeviceVector<Real> d_stencil_coef_y;
-    amrex::Gpu::DeviceVector<Real> d_stencil_coef_z;
-    if (maxwell_solver_id == MaxwellSolverAlgo::PSATD) {
-        const int fg_nox = WarpX::field_gathering_nox;
-        const int fg_noy = WarpX::field_gathering_noy;
-        const int fg_noz = WarpX::field_gathering_noz;
-
-        // Compute real-space stencil coefficients along x
-        amrex::Vector<Real> h_stencil_coef_x = getFornbergStencilCoefficients(fg_nox, false);
-        d_stencil_coef_x.resize(h_stencil_coef_x.size());
-        amrex::Gpu::copyAsync(amrex::Gpu::hostToDevice,
-                              h_stencil_coef_x.begin(),
-                              h_stencil_coef_x.end(),
-                              d_stencil_coef_x.begin());
-
-        // Compute real-space stencil coefficients along y
-        amrex::Vector<Real> h_stencil_coef_y = getFornbergStencilCoefficients(fg_noy, false);
-        d_stencil_coef_y.resize(h_stencil_coef_y.size());
-        amrex::Gpu::copyAsync(amrex::Gpu::hostToDevice,
-                              h_stencil_coef_y.begin(),
-                              h_stencil_coef_y.end(),
-                              d_stencil_coef_y.begin());
-
-        // Compute real-space stencil coefficients along z
-        amrex::Vector<Real> h_stencil_coef_z = getFornbergStencilCoefficients(fg_noz, false);
-        d_stencil_coef_z.resize(h_stencil_coef_z.size());
-        amrex::Gpu::copyAsync(amrex::Gpu::hostToDevice,
-                              h_stencil_coef_z.begin(),
-                              h_stencil_coef_z.end(),
-                              d_stencil_coef_z.begin());
-
-        amrex::Gpu::synchronize();
-    }
```

Comment:
The computation and host/device storage of the finite-order centering stencil coefficients is now done only once at initialization (see Source/WarpX.cpp) and not every time the function `WarpX::UpdateAuxilaryDataStagToNodal` is called.